bin/http-signature-header-js

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env node
+
+const program = require('commander');
+const httpSigs = require('../lib/');
+const fs = require('fs');
+const util = require('util');
+const httpMessageParser = require('http-message-parser');
+
+// this can be use to await paths to private keys and other files.
+// also can get be used to dereference keyId in middleware
+const readFile = util.promisify(fs.readFile);
+
+/**
+ * Simple wrapper around node's process.stdin.
+ * This allows us to get the result of cat and other commands.
+ *
+ * @param {string} [encoding='utf8'] - The http message's encoding.
+ *
+ * @returns {Promise<string>} Should return the http message.
+ */
+async function getStdin(encoding = 'utf8') {
```

Comment:
If utf8 alone is good enough then consider this: https://www.npmjs.com/package/get-stdin

Reply:
@msporny directly said not to use that plugin (although it basically does the same thing with a few extra caveats). We don't really need it though node's stdin events are pretty easy to wrap a promise around.

Reply:
I don't think we'll be processing *just* utf8... so what's there is good, AFAICT.